Modules/Synchronization/src/Synchronization.Domain/Entities/Sync/ExternalEvent.cs

```diff
@@ -44,6 +43,11 @@ public ExternalEvent(ExternalEventType type, IdentityAddress owner, long index,
     public SyncRunId? SyncRunId { get; private set; }
     public IReadOnlyCollection<SyncError> Errors => _errors;
 
+    public void UpdateIndex(long newIndex)
+    {
+        Index = newIndex;
+    }
+
```

Comment:
Hint for the reviewer: this method is necessary now, because we cannot recreate the External Event in the SynchronizationDbContext anymore (because the generic SynchronizationDbContext.CreateExternalEvent method doesn't know which external event to create).

But even though letting the Index default to `0` in the cosntructor is kind of ugly, I still think this apporach is somewhat better than the old one, because now the SynchronizationDbContext.CreateExternalEvent is way simpler than before.